File: JournalForge/Models/AppSettings.cs
namespace JournalForge.Models;

public class AppSettings
{
    /// <summary>
    /// OpenAI API Key. Set this to your OpenAI API key to enable AI-powered features.
    /// Get your API key from: https://platform.openai.com/api-keys
    /// </summary>
    public string OpenAIApiKey { get; set; } = string.Empty;
    
    /// <summary>
    /// OpenAI Model to use. Default is "gpt-4o-mini" for cost-effectiveness.
    /// Other options: "gpt-4o", "gpt-4-turbo", "gpt-3.5-turbo"
    /// </summary>
    public string OpenAIModel { get; set; } = "gpt-4o-mini";
}

File: android-app/app/src/main/java/com/journalforge/app/models/AppSettings.kt
package com.journalforge.app.models

import com.google.gson.annotations.SerializedName

/**
 * Application settings
 */
data class AppSettings(
    @SerializedName("openAIApiKey")
    val openAIApiKey: String? = null,
    
    @SerializedName("openAIModel")
    val openAIModel: String = "gpt-4o-mini",
    
    @SerializedName("firebaseApiKey")
    val firebaseApiKey: String? = null,
    
    @SerializedName("googleClientId")
    val googleClientId: String? = null
)
